base/MAPL_ISO8601_Time.F90

```diff
@@ -0,0 +1,720 @@
+      integer, dimension(DIM_DF), intent(out) :: fields
+      integer, intent(inout) :: stat
+      integer, dimension(DIM_DF) :: f
+      if(is_delimited_date(s)) then
+         call parse_isostring(s, fields, FDD, stat)
+      else if(is_undelimited_date(s)) then
+         call parse_isostring(s, fields, FUD, stat)
+      else
+         stat = -1
+      end if  
+   end subroutine parse_datestring
+
+   subroutine parse_timestring(s, fields, stat)
+      character(len=*), intent(in) :: s
+      integer, dimension(DIM_TF), intent(out) :: fields
+      integer, intent(inout) :: stat
+
+      fields(size(fields)) = TIMEZONE_DEFAULT
+
+      if(is_delimited_time(s)) then
+         call parse_isostring(s, fields(1:4), FDT, stat)
+      else if(is_undelimited_time(s)) then
+         call parse_isostring(s, fields(1:4), FUT, stat)
+      else
+         stat = -1
+      end if
+   end subroutine parse_timestring
+
+   subroutine parse_datetimestring(s, datefields, timefields, stat)
+      character(len=*), intent(in) :: s
+      integer, dimension(DIM_DF), intent(out) :: datefields
+      integer, dimension(DIM_TF), intent(out) :: timefields
+      integer, intent(inout) :: stat
+      character(len=len(s)), dimension(2) :: parts 
+ 
+      parts = split_datetime_string(s) 
+      call parse_datestring(trim(parts(1)),datefields,stat)
+      if(stat == 0) then
+         call parse_timestring(trim(parts(2)),timefields,stat)
+      else
+         stat = -1
+      endif
+   end subroutine parse_datetimestring
+   
+! END LOW-LEVEL PARSERS
+
+
+! LOW-LEVEL DATE & TIME PROCESSING UTILITIES
+   ! Return true if factor divides dividend evenly, false otherwise
+   pure logical function is_factor(dividend, factor)
+      integer, intent(in) :: dividend
+      integer, intent(in) :: factor
+      ! mod returns the remainder of dividend/factor, and if it is 0, factor divides dividend evenly
+      if(factor /= 0) then ! To avoid divide by 0
+          is_factor = mod(dividend, factor)==0
+      else
+          is_factor = .false.
+      endif
+   end function is_factor
+
+   ! Return true if factor does not divide dividend evenly, false otherwise
+   ! see is_factor
+   pure logical function is_not_factor(dividend, factor)
+      integer, intent(in) :: dividend
+      integer, intent(in) :: factor
+      is_not_factor = .not. is_factor(dividend, factor)
+   end function is_not_factor
+
+   pure logical function is_between(lower, upper, n) 
+      integer, intent(in) :: lower
+      integer, intent(in) :: upper
+      integer, intent(in) :: n
+      is_between = n > lower .and. n < upper
+   end function is_between
+
+   ! Return true if y is a leap year, false otherwise
+   pure logical function is_leap_year(y)
+      integer, intent(in) :: y
+      ! Leap years are years divisible by 400 or (years divisible by 4 and not divisible by 100)
+      is_leap_year = is_factor(y,400) .or. ( is_factor(y, 4) .and. is_not_factor(y, 100) )
+   end function is_leap_year
```

Comment:
This function occurs in other contexts.   Not sure where it is, but probably best that we find it and use it to avoid duplication.   Your version is probably better than the existing one, as it uses `pure`.